kolibri/plugins/document_epub_render/assets/src/views/LoadingScreen.vue

```diff
@@ -32,9 +43,14 @@
 
   .loading-screen {
     position: absolute;
-    top: 50%;
-    left: 50%;
-    transform: translate(-50%, -50%);
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    z-index: 7;
+    display: flex;
+    align-items: center;
+    justify-content: center;
```

Comment:
We've tried to avoid using flexbox due to [less-than-complete support](https://caniuse.com/#feat=flexbox) on older Android browsers

Reply:
Understood. This should be just fine though, unless you feel strongly this should be changed.

Reply:
> This should be just fine though

could you elaborate?

I don't actually know a lot about what parts of flexbox worked on older android browsers, so I've generally just advocated avoiding it. However if it works in some specific cases, we can adjust our conventions accordingly but we'll need to make sure everyone on the team understands when it's permitted.

Reply:
The styles in Kolibri run through `autoprefixer` correct? That's what I've seen thus far. With prefixed styles here, it should work just fine.

Some gotchas are with `order`, `justify-content: space-around`, `flex-grow`, and worst of all in my experience, `flex-direction: column`. 

https://benfrain.com/flexbox-gotchas-for-android-2-1-4-3-stock-browser/
https://blog.frankmtaylor.com/2013/10/11/true-story-the-android-web-browser-is-the-honey-badger-of-flexbox/

Reply:
Also, if the Android Browser is really supported, I think we should update the list here: https://kolibri.readthedocs.io/en/latest/access.html#compatibility

Reply:
Yup, the styles get auto-prefixed.

I think it was those two articles that made me decide to simply ban `flexbox` from our codebase!

Basically the thinking was that simpler, more conservative conventions are easier to enforce. I'm open to reversing that (especially if it turns out it applies to browsers we don't actually need to support), but my guess is that maybe it's helped us avoid browser-specific layout bugs?

I created a new issue to resolve the browser support question here: https://github.com/learningequality/kolibri-docs/issues/95

Reply:
Personally, I don't think that's best to ban flexbox. If you continue with the approach of banning CSS syntaxes, you'll end up banning everything. For instance, iOS Safari has some pretty annoying issues with fixed positioning that haven't been fixed as far as I know-- and that's a current browser. So would you ban all fixed positioning based off that? I would say no. And if all developers do that now or did that years ago, web development and design would have not progressed because no one would use new abilities.

In this instance, I truly believe no issues will be present, based off my experience. I think in these scenarios where there is a question of whether this will work correctly, you can take into account the use and what happens when it breaks. In this case, it's a loading animation and if the flexbox doesn't work, it simply displays in the top left corner. Now if we're talking a page template/layout, that's something entirely different-- the fallback is probably a broken experience.

Reply:
Yeah I totally trust that this specific case is fine.

My main motivation is about the precedent we set for more junior devs or drive-by contributors: This sort of ban has been a cheaper alternative to rigorously testing every PR in every browser.

We've tended to use caniuse.com as the "source of truth" to guide us. At present, we don't have many instances of intentional 'graceful degradation' in the app - though I'm sure there are places of unintentional degradation, graceful or not!

For what it's worth, on Studio flexbox is allowed along with css variables, and css grid, and other more modern features. My general attitude with Kolibri has been much more conservative because of the unique target audience.

Maybe we should take the temperature from the team at some point and see if this kind of blanket ban is more annoying or more helpful?

Reply:
Well, if Can I Use is the source of truth and lists it as "partial support" with prefixes, I think you'll need to provide more criteria on what goes and what doesn't. At the moment, it seems a bit arbitrary to me.

There may be a bigger flexbox problem than you're aware... the Pure CSS used for `<KGrid>` uses flexbox!

Reply:
right - full support with prefixes was what I meant.

I believe purecss falls back to inline blocks when flexbox isn't available. Though looking at their github page they say Android 4.4+ which is a big surprise!

Reply:
I'll 'resolve' this thread since it's sort of tangential to this PR. We can either keep it or not, but some follow-up in https://github.com/learningequality/kolibri-docs/issues/95 is definitely warranted